drivers/net/ethernet/intel/idpf/idpf_eth_idc.h

```diff
@@ -0,0 +1,145 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/* Copyright (C) 2024 Intel Corporation */
+
+#ifndef _IDPF_ETH_IDC_H_
+#define _IDPF_ETH_IDC_H_
+
+#include <linux/auxiliary_bus.h>
+
+struct idpf_vc_xn_params;
+
+enum idpf_eth_idc_event_code {
+	/* Following events are from main driver to aux/eth driver */
+	IDPF_ETH_IDC_EVENT_LINK_CHANGE,
+	IDPF_ETH_IDC_EVENT_RESET_INITIATED,
+	IDPF_ETH_IDC_EVENT_RESET_COMPLETE,
+	IDPF_ETH_IDC_EVENT_RELEASE_RESOURCES,
+	IDPF_ETH_IDC_EVENT_POST_INIT,
+
+	/* Following requests are from auxiliary eth driver to main driver */
+	IDPF_ETH_IDC_EVENT_REQ_HARD_RESET,
+};
+
+/**
+ * struct idpf_eth_idc_auxiliary_dev_caps - Capabilities info
+ * @csum_caps: See enum virtchnl2_cap_txrx_csum
+ * @seg_caps: See enum virtchnl2_cap_seg
+ * @hsplit_caps: See enum virtchnl2_cap_rx_hsplit_at
+ * @rsc_caps: See enum virtchnl2_cap_rsc
+ * @rss_caps: See enum virtchnl2_cap_rss
+ * @other_caps: See enum virtchnl2_cap_other
+ * @max_tx_hdr_size: Max header length hardware can parse/checksum, in bytes.
+ * @max_sg_bufs_per_tx_pkt: Max number of scatter gather buffers that can be
+ *			    sent per transmit packet without needing to be
+ *			    linearized.
+ * @min_sso_packet_len: Min packet length supported by device for single
+ *	segment offload
+ * @q_info: Max queue information
+ * @crc_enable: Enable CRC insertion offload
+ */
+struct idpf_eth_idc_auxiliary_dev_caps {
+	__le32 csum_caps;
+	__le32 seg_caps;
+	__le32 hsplit_caps;
+	__le32 rsc_caps;
+	__le64 rss_caps;
+	__le64 other_caps;
+	__le16 max_tx_hdr_size;
+	u8 max_sg_bufs_per_tx_pkt;
+	u8 min_sso_packet_len;
+	struct idpf_max_q q_info;
+	bool crc_enable;
+};
+
+/**
+ * struct idpf_eth_idc_event - ethernet idc events
+ * @event_data: Event data
+ * @event_code: Event code
+ */
+struct idpf_eth_idc_event {
+	void *event_data;
+	enum idpf_eth_idc_event_code event_code;
+};
+
+struct idpf_eth_idc_dev_info;
```

Comment:
See if we can avoid the forward declaration

Reply:
We agreed to this one, it is needed as it because of the dependency.